lib/practice_exercises.rb

```diff
@@ -1,13 +1,54 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) becuase the it depends on the array.length
+# Space Complexity: O(n) because we created new array
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  array = string.chars
+  orginal_char_array = array.select do |char|
+    /[A-Za-z]/.match char
+  end
+  
+  char_array = array.select do |char|
+    /[A-Za-z]/.match char
+  end
+  length = char_array.length
+  first_index = 0
+  last_index = length -1
+  while first_index < last_index
+    temp = char_array[first_index]
+    char_array[first_index] = char_array[last_index]
+    char_array[last_index] = temp
+    first_index += 1
+    last_index -= 1
+  end
+  
+  if char_array.join.upcase == orginal_char_array.join.upcase
+    return true
+  else
+    return false
+  end
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+
+# Time Complexity: O(n) depends on min.length
+# Space Complexity: O(1) no new array created
+
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍